packages/data/src/redux-store/index.js

```diff
@@ -631,7 +646,27 @@ function mapSelectorWithResolver(
 					metadataActions.failResolution( selectorName, args, error )
 				);
 			}
-		}, 0 );
+		}
+
+		queue.push( [ startResolution, fulfillResolution ] );
+
+		window.queueMicrotask( () => {
```

Comment:
btw this code should be able to run also in raw Node.js, without any DOM APIs. `Promise.resolve().then( ... )` is a universal way to queue a microtask.

Reply:
Also, when starting 100 resolutions at once, there will be 100 microtasks queued of which 99 will see an empty queue. Could we queue just one microtask?

Reply:
How do we know when to queue that one microtask? We have some similar techniques across the codebase, but I haven't seen one where only one task is queued, and not sure how it'd be possible?

Reply:
Then it would push a task to the queue and will still be done?

Reply:
Depends on how exactly you loop through the queue array. Your `for ( const fulfill of queue )` will see new items added to `queue` inside the loop. But `queue.forEach` won't, it will process only items that were in the queue when the `.forEach` was called.